app/src/main/java/ch/sdp/vibester/helper/IntentSwitcher.kt

```diff
@@ -2,12 +2,18 @@ package ch.sdp.vibester.helper
 
 import android.content.Context
 import android.content.Intent
-import ch.sdp.vibester.activity.WelcomeActivity
+import java.io.Serializable
 
 class IntentSwitcher private constructor() {
     companion object {
-        fun switchBackToWelcome(ctx: Context) {
-            ctx.startActivity(Intent(ctx, WelcomeActivity::class.java))
+        fun switch(ctx: Context, arg: Class<*>?, extrasMap: Map<String, Serializable>?) {

```

Comment:
we can put extrasmap as null by default, so we dont seutp null in each call of welcome activity

Reply:
yes